lona/pytest.py

```diff
@@ -6,31 +13,38 @@
 from lona import LonaApp
 
 
+@dataclass
 class LonaContext:
-    def __init__(self, client, app, server):
-        self.client = client
-        self.app = app
-        self.server = server
+    client: TestClient
+    app: None | LonaApp
+    server: LonaServer
 
-    def make_url(self, path=''):
+    def make_url(self, path: str = '') -> str:
         return str(self.client.make_url(path))
 
 
 @pytest.fixture()
 def lona_app_context(request, aiohttp_client):
-    async def setup_lona_app_context(setup_app, project_root=''):
+    async def setup_lona_app_context(
+            setup_app: Callable[[LonaApp], None],
+            project_root: str = '',
+    ) -> LonaContext:
         # setup lona app
-        lona_app = LonaApp(project_root or request.fspath)
+        lona_app = LonaApp(project_root or str(request.fspath))
         setup_app(lona_app)
 
         # setup client
-        def setup_aiohttp_app(loop):
+        def setup_aiohttp_app(loop: AbstractEventLoop) -> Application:
             lona_app.setup_server(loop=loop)
 
+            assert lona_app.aiohttp_app is not None
```

Comment:
Why is this necessary? This has nothing to todo with types also, so it should be mentioned in the commit message at least.

This assert can fail in end user test suites. Therefore it should have an assertion message

Reply:
> Why is this necessary?

That's a good question. Because from types perspective it is `Optional[Application]`, but should be not optional.
Before `lona_app.setup_server()` it is `None` and become real application after that.

I removed assertions and used `cast` instead.